app/src/main/java/ca/wheresthebus/ui/trips/TripsFragment.kt

```diff
@@ -22,21 +37,82 @@ class TripsFragment : Fragment() {
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        val dashboardViewModel =
-            ViewModelProvider(this).get(TripsViewModel::class.java)
+        tripsViewModel = ViewModelProvider(this)[TripsViewModel::class.java]
 
         _binding = FragmentTripsBinding.inflate(inflater, container, false)
         val root: View = binding.root
 
-        val textView: TextView = binding.textDashboard
-        dashboardViewModel.text.observe(viewLifecycleOwner) {
-            textView.text = it
-        }
+
+        mainDBViewModel = ViewModelProvider(requireActivity())[MainDBViewModel::class]
+
+        setUpAdapter()
         return root
     }
 
+    private fun setUpAdapter() {
+        val currentTime = LocalDateTime.now()
+
+        val trips = mainDBViewModel.getTrips()
+            .sortedBy { it.getClosestTime(currentTime) }
+            .groupBy { trip ->
+                when {
+                    trip.isActive(currentTime) -> TripType.ACTIVE
+                    trip.isToday(currentTime) -> TripType.TODAY
+                    else -> TripType.INACTIVE
+                }
+            }
+
+        activeTripsView = binding.recyclerActiveTrips
+        inactiveTripsView = binding.recyclerInactiveTrips
+        upcomingTripsView = binding.recyclerUpcomingTrips
+
+        trips[TripType.ACTIVE].orEmpty().let {
+            if (it.isEmpty()) {
+                binding.labelActive.visibility = View.GONE
+            }
+            activeTripAdapter = TripAdapter(it)
+
+            activeTripsView.apply {
+                layoutManager = LinearLayoutManager(context)
+                adapter = activeTripAdapter
+            }
+        }
+
+        trips[TripType.TODAY].orEmpty().let {
+            if (it.isEmpty()) {
+                binding.labelUpcomingTrips.visibility = View.GONE
+            }
+            upcomingTripAdapter = TripAdapter(it)
+
+            upcomingTripsView.apply {
+                layoutManager = LinearLayoutManager(context)
+                adapter = upcomingTripAdapter
+            }
+        }
+
+        trips[TripType.INACTIVE].orEmpty().let {
+            if (it.isEmpty()) {
+                binding.labelAllTrips.visibility = View.GONE
+            }
+            inactiveTripAdapter = TripAdapter(it)
+
+            inactiveTripsView.apply {
+                layoutManager = LinearLayoutManager(context)
+                adapter = inactiveTripAdapter
+            }
+        }
+    }
+
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
+
+    companion object {
+        object TripType {
+            const val ACTIVE = "active"
```

Comment:
there seems to be a lot of ACTIVE / INACTIVE constants being made throughout the program, could make have a Globals file?